server/src/events.mjs

```diff
@@ -87,48 +97,112 @@ function getRegisteredEventListener(userId, method) {
   return eventListenerMap[userId][method];
 }
 
-// Remove mapping from event listener request method name from our map
-// Attempts to send events to this listener going forward will fail
-function deregisterEventListener(userId, oMsg, ws) {
-  if (!eventListenerMap[userId] || !eventListenerMap[userId][oMsg.method]) {
+/**
+ * Removes the mapping from the event listener request method name from eventListenerMap. 
+ * Attempts to send events to this listener going forward will fail.
+ * @param {string} userId - The ID of the user.
+ * @param {Object} metadata - An object containing metadata for the event.
+ * @param {WebSocket} ws - The WebSocket object.
+ * @returns {void}
+*/
+function deregisterEventListener(userId, metadata, ws) {
+  const { event } = metadata;
+  if (!eventListenerMap[userId] || !eventListenerMap[userId][event]) {
     return;
   }
 
-  const wsArr = eventListenerMap[userId][oMsg.method].wsArr;
+  const wsArr = eventListenerMap[userId][event].wsArr;
   const wsIndex = wsArr.findIndex((item) => item === ws);
 
   if (wsIndex !== -1) {
     wsArr.splice(wsIndex, 1);
-    logger.debug(`Deregistered event listener mapping: ${userId}:${oMsg.method}`);
+    logger.debug(`Deregistered event listener mapping: ${userId}:${event}`);
   }
 
   if (wsArr.length === 0) {
-    delete eventListenerMap[userId][oMsg.method];
+    delete eventListenerMap[userId][event];
+  }
+}
+
+/**
+ * Extracts event data from a given message object based on provided configuration.
+ * @param {object} oMsg - The message object to extract event data from.
+ * @param {object} config - The configuration object for the event data extraction.
+ * @param {boolean} isOn - Whether eventListener enable or disable request
+ * @returns {object | false} - An object containing the extracted event data or false if the extraction fails.
+*/
+function extractEventData(oMsg, config, isOn) {
+  const { searchRegex, method } = config;
+
+  if (!new RegExp(searchRegex).test(JSON.stringify(oMsg))) {
+    return false;
+  }
+
+  const extractedMethod = JSONPath.query(oMsg, method);
+
+  if (extractedMethod.length === 0) {
+    return false;
+  }
+
+  const methodName = extractedMethod[0];
+  const metadata = {
+    registration: {},
+    event: methodName,
```

Comment:
This goes against the design.

metadata.event should contain the entire event payload that's being processed. We should be able to retrieve the method being followed through eventListenerMap[userId][method]

Reply:
Note that I'm on-board with adding a "method" field to the metadata and putting it there if it makes life easier.